.github/workflows/build-go.yaml

```diff
@@ -11,7 +11,7 @@ on:
 
 env:
   GO111MODULE: on
-  GO_VERSION: "^1.18"
+  GO_VERSION: "^1.21"
```

Comment:
The build and release workflows were [already building with Go `1.22.3`](https://github.com/bitwarden/sdk/actions/runs/9352811045/job/25741824418?pr=816#step:3:20) prior to this change. This is merely updating the workflows to match the minimum Go versions supported in our module.